FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Models/Jobs/FilterInfo.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Synapse.Common.Models.FhirSearch;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Synapse.Common.Models.Jobs
+{
+    /// <summary>
+    /// The filter information of a job, which is immutable once a job is created.
+    /// </summary>
+    public class FilterInfo
+    {
+        public FilterInfo(
+            FilterScope filterScope,
+            string groupId,
+            DateTimeOffset since,
+            IEnumerable<TypeFilter> typeFilters,
+            IEnumerable<string> processedPatientIds)
+        {
+            FilterScope = filterScope;
```

Comment:
Same here, since the `typeFilters` and `processedPatientIds` are checked with a "default value for null input" strategy, it is also necessary to check or validate other parameters like `groupId`?

Reply:
we allow `groupId` to be null for `system` filter scope, so we don't need to check it here, I will add group id null-or-whitespace check when using it in `GetGroupMembersAsync()`.

For other parameters `filterScope` and `since`, they are enum/DateTimeOffset type, null is not allowed when calling it.